utils/database.ts

```diff
@@ -22,14 +23,22 @@ export function getConnection(
 ): TaskEither<Error, Connection> {
   // Use an existing connection inside the connection manager.
   const maybeActiveConnection = getConnectionManager().connections[0];
-  if (maybeActiveConnection) {
+  if (maybeActiveConnection && maybeActiveConnection.isConnected) {
     return taskEither.of(maybeActiveConnection);
   }
+  if (maybeActiveConnection) {
+    // If the existing connection is disconnected
+    // we perform a reconnection and return it
+    return tryCatch(
+      () => maybeActiveConnection.connect(),
+      err => new Error(`Error during postgres re-connection [err:${err}]`)
+    );
+  }
```

Comment:
maybe we can refactor this and collapse into a single chain :)

Reply:
Fixed in df37e49